app/geo/gui/src/ts/twxplore/gui/geo/components/filterSliders/FilterComponent.tsx

```diff
@@ -0,0 +1,243 @@
+import {MapStringFeatureAttributeState} from "../../states/map/MapFeatureAttributeState/MapStringFeatureAttributeState";
+import {FormControl, TextField} from "@material-ui/core";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      width: 300,
+    },
+    margin: {
+      height: theme.spacing(3),
+    },
+    formControl: {
+      margin: theme.spacing(1),
+      minWidth: 120,
+      maxWidth: 300,
+    },
+    chips: {
+      display: "flex",
+      flexWrap: "wrap",
+    },
+    chip: {
+      margin: 2,
+    },
+    noLabel: {
+      marginTop: theme.spacing(3),
+    },
+  })
+);
+
+function valuetext(value: number) {
+  return `${value}`;
+}
+
+//var filterIndexOfAttribute: number | null; //necessary to pass this into onChangeCommitted since it
+const FilterSlidersImpl: React.FunctionComponent<{featureType: string}> = ({
+  featureType,
+}) => {
+  const classes = useStyles();
+  const state: MapState = useSelector(
+    (rootState: RootState) => rootState.app.map
+  );
+
+  const setInitialFilter = (
+    filterIndexOfAttribute: number,
+    attributeName: string,
+    stateOfAttribute: MapFeatureAttributeState
+  ) => {
+    /*dispatch(
+      setFilter(
+        filterIndexOfAttribute,
+        "domain",
+        getFeatureAttributeByName(attributeName).fieldType
+      )
+    );*/
+    dispatch(setFilter(filterIndexOfAttribute, "name", attributeName));
+    dispatch(
+      setFilter(
+        filterIndexOfAttribute,
+        "fieldType",
+        getFeatureAttributeByName(attributeName).fieldType
+      )
+    );
+    dispatch(
+      setFilter(
+        filterIndexOfAttribute,
+        "type",
+        getFeatureAttributeByName(attributeName).filterType
+      )
+    );
+    switch (getFeatureAttributeByName(attributeName).typeOf) {
+      case TypeOfFeatureAttribute.NUMBER: {
+        stateOfAttribute = stateOfAttribute as MapNumericFeatureAttributeState;
+        dispatch(
+          setFilter(filterIndexOfAttribute, "value", [
+            stateOfAttribute.min,
+            stateOfAttribute.max,
+          ])
+        );
+        break;
+      }
+      //Do nothing, we don't populate the filter until the user makes a selection
+      case TypeOfFeatureAttribute.STRING: {
+        stateOfAttribute = stateOfAttribute as MapStringFeatureAttributeState;
+
+        break;
+      }
+      default: {
+        throw Error("Unhandled case for typeOf FeatureAttribute");
+      }
+    }
+  };
+
+  const returnFilterComponent = (
+    filterIndexOfAttribute: number,
+    attributeName: string,
+    stateOfAttribute: MapFeatureAttributeState
+  ) => {
+    switch (getFeatureAttributeByName(attributeName).typeOf) {
+      case TypeOfFeatureAttribute.NUMBER: {
+        stateOfAttribute = stateOfAttribute as MapNumericFeatureAttributeState;
+        return (
```

Comment:
Following from the above, you could do:
featureAttribute.filterType.render(stateOfAttribute, ...) -> React.ReactNode
But I think I prefer the way you have it right now, keeping the actual JSX in one place.